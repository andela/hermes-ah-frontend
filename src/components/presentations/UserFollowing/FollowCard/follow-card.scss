@import "../../../../assets/stylesheets/index.scss";

p {
  font-family: $font-stack;
}
.main-card {
  display: flex;
  justify-content: space-between;
  text-align: left;
  width: 95%;
  margin: 10px;
  padding: 10px 45px;
  float: left;
  overflow: auto;
  border: solid 1px #e8e7e7;
  border-radius: 4px;
  box-shadow: 0 3px 6px #bfbfbf;

  .details {
    display: flex;
  }

  .img-section {
    width: 70px;
    height: 70px;
    margin: 0;
    background-color: #808080;
    border-radius: 66px;
    float: left;
    text-align: center;
    color: #ffffff;

    .initials {
      margin-top: 8px;
      font-size: 41px;
    }

    .profile-pic {
      width: 70px;
      height: 70px;
      border-radius: 50%;
    }
  }

  .highlights {
    margin: 0;
    padding: 0 10px;

    .name {
      font-size: 18px;
    }

    p {
      font-size: 14px;
    }
  }

  .button-div {
    padding: 25px 0;

    .btn-following {
      background: #fff;
      text-transform: capitalize;
      padding: 5px 10px;
      border-radius: 10px;
      border: solid 1px $primary-color;
      color: $primary-color;
      font-size: 14px;
      cursor: pointer;
    }

    .btn-follower {
      background: $primary-color;
      text-transform: capitalize;
      padding: 5px 10px;
      border-radius: 10px;
      border: solid 1px #fff;
      color: #fff;
      font-size: 14px;
      cursor: pointer;
    }

    .btn-following:hover{
      background: $primary-color;
      color: #fff;
    }

    .btn-follower:hover{
      background: #fff;
      color: $primary-color;
      border: solid 1px $primary-color;
    }
  }
}

.grid-render {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 19px;
}

@media screen and (max-width: 768px) {
  .main-card {
    flex-direction: column;
    justify-content: center;
    text-align: center;
    padding: 10px;

    .details {
      flex-direction: column;
    }

    .img-section {
      margin: 20px auto;
    }
  }
}

@media screen and (max-width: 480px) {
  .grid-render {
    grid-template-columns: repeat(1, 1fr);
  }
}
