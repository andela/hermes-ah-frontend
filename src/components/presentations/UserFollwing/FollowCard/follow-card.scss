@import "../../../../assets/stylesheets/index.scss";

p {
  font-family: $font-stack;
}
.main-card {
  text-align: left;
  width: 95%;
  margin: 20px auto;
  padding: 10px 20px;
  float: left;
  overflow: auto;
  border: solid 1px $secondary-color;
  border: solid 1px #959595;
  border-radius: 8px;
  box-shadow: 0 1px 6px 0 #888888;

  .img-section {
    width: 13%;
    height: 80px;
    margin: 0 10px;
    background-color: grey;
    border-radius: 66px;
    float: left;
    text-align: center;
    color: white;

    .initials {
      margin-top: 8px;
      font-size: 41px;
    }

    .profile-pic {
      width: 100%;
      height: 80px;
      border-radius: 50%;
    }
  }

  .highlights {
    width: 50%;
    margin: 0 10px;
    padding: 0 10px;
    float: left;

    .name {
      font-size: 18px;
    }

    p {
      font-size: 12px;
    }
  }

  .button-div {
    width: 15%;
    float: right;
    text-align: right;

    .btn-following {
      background: #fff;
      text-transform: capitalize;
      margin-top: 20px;
      padding: 4px;
      border-radius: 10px;
      width: 104%;
      border: solid 1px $primary-color;
      color: $primary-color;
      font-size: 19px;
    }

    .btn-follower {
      background: $primary-color;
      text-transform: capitalize;
      margin-top: 20px;
      padding: 4px;
      border-radius: 10px;
      width: 104%;
      border: solid 1px #fff;
      color: #fff;
      font-size: 19px;
    }

    .btn-following:hover{
      background: $primary-color;
      color: #fff;
    }

    .btn-follower:hover{
      background: #fff;
      color: $primary-color;
      border: solid 1px $primary-color;
    }
  }
}

.grid-render {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 19px;
}
