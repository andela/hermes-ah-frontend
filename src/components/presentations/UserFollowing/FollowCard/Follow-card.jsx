import React from 'react';
import PropTypes from 'prop-types';
import './follow-card.scss';

const Follow = ({
  initials,
  imageUrl,
  name,
  bio,
  button,
  btnClass,
  Id,
  buttonEvent,
}) => {
  return (
    <div className="main-card">
      <div className="details">
        <div className="img-section">
          {imageUrl ? (
            <img className="profile-pic" src={imageUrl} alt="" />
          ) : (
            <p className="initials">{initials}</p>
          )}
        </div>
        <div className="highlights">
          <p className="name">{name}</p>
          <p>{bio}</p>
        </div>
      </div>
      <div className="button-div">
        <button
          type="submit"
          id={Id}
          onClick={buttonEvent}
          className={btnClass}
        >
          {button}
        </button>
      </div>
    </div>
  );
};

Follow.propTypes = {
  initials: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  button: PropTypes.string.isRequired,
  btnClass: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  Id: PropTypes.string.isRequired,
  buttonEvent: PropTypes.func.isRequired,
};

export default Follow;
