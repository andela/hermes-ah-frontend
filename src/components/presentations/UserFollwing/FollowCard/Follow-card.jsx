import React from 'react';
import PropTypes from 'prop-types';
import './follow-card.scss';

const Follow = ({ initials, imageUrl, name, bio, button, btnClass }) => {
  return (
    <div className="main-card">
      <div className="img-section">
        {imageUrl ? (
          <img className="profile-pic" src={imageUrl} alt="" />
        ) : (
          <p className="initials">{initials}</p>
        )}
      </div>
      <div className="highlights">
        <p className="name">{name}</p>
        <p>{bio}</p>
      </div>
      <div className="button-div">
        <button type="submit" className={btnClass}>
          {button}
        </button>
      </div>
    </div>
  );
};

Follow.propTypes = {
  initials: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  button: PropTypes.string.isRequired,
  btnClass: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
};

export default Follow;
