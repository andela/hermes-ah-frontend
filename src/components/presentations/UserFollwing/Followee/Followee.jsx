import React from 'react';
import PropTypes from 'prop-types';
import FollowingCard from '../FollowCard/Follow-card';
import '../FollowCard/follow-card.scss';

const FolloweeList = ({ userFollowee }) => {
  const { userFollowee: followee } = userFollowee;
  return (
    <div className="grid-render">
      {followee.map(user => (
        <FollowingCard
          key={user.id}
          imageUrl={user.follower.image_url}
          initials={`${user.follower.first_name
            .charAt(0)
            .toUpperCase()}${user.follower.last_name.charAt(0).toUpperCase()}`}
          name={`${user.follower.first_name} ${user.follower.last_name}`}
          bio={`${user.follower.bio}`}
          button="follow"
          btnClass="btn-follower"
        />
      ))}
    </div>
  );
};

FolloweeList.propTypes = {
  userFollowee: PropTypes.shape().isRequired,
};

export default FolloweeList;
