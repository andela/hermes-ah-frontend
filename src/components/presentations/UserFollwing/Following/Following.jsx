import React from 'react';
import PropTypes from 'prop-types';
import FollowingCard from '../FollowCard/Follow-card';
import '../FollowCard/follow-card.scss';

const FollowingList = ({ userFollowing }) => {
  const { userFollowing: following } = userFollowing;
  return (
    <div className="grid-render">
      {following.map(user => (
        <FollowingCard
          key={user.id}
          imageUrl={user.followee.image_url}
          initials={`${user.followee.first_name
            .charAt(0)
            .toUpperCase()}${user.followee.last_name.charAt(0).toUpperCase()}`}
          name={`${user.followee.first_name} ${user.followee.last_name}`}
          bio={`${user.followee.bio}`}
          button="following"
          btnClass="btn-following"
        />
      ))}
    </div>
  );
};

FollowingList.propTypes = {
  userFollowing: PropTypes.shape().isRequired,
};

export default FollowingList;
