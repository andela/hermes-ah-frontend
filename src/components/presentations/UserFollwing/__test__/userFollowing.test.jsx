import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Followee from '../Followee/Followee';
import Following from '../Following/Following';
import FollowCard from '../FollowCard/Follow-card';

const userFollowee = {
  imageUrl: '',
  initials: '',
  name: '',
  bio: '',
  button: '',
  btnClass: '',
};

const userFollowing = {
  imageUrl: '',
  initials: '',
  name: '',
  bio: '',
  button: '',
  btnClass: '',
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ userFollowee, userFollowing });

const UserFolloweeComponent = (
  <Provider store={store}>
    <Followee userFollowee={{ userFollowee: [{}] }} />
  </Provider>
);

const UserFollowingComponent = (
  <Provider store={store}>
    <Following userFollowing={{ userFollowing: [{}] }} />
  </Provider>
);

describe('<Followee />', () => {
  it('should render user followee page', () => {
    const wrapper = shallow(UserFolloweeComponent);
    expect(wrapper.find('div'));
  });
});

describe('<Following />', () => {
  it('should render user following page', () => {
    const wrapper = shallow(UserFollowingComponent);
    expect(wrapper.find('div'));
  });
});

describe('<FollowCard />', () => {
  it('should render user follow card', () => {
    const wrapper = shallow(
      <FollowCard
        imageUrl=""
        initials=""
        name=""
        bio=""
        button=""
        btnClass=""
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
