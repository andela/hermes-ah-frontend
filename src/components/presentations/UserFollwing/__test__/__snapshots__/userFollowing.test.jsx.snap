// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowCard /> should render user follow card 1`] = `ShallowWrapper {}`;
